src/app/application.cpp

```diff
@@ -403,15 +403,63 @@ void Application::processParams(const QStringList &params)
         return;
     }
 #endif
+    BitTorrent::AddTorrentParams torrentParams;
+    bool skipTorrentDialog = false;
 
     foreach (QString param, params) {
         param = param.trimmed();
+
+        // Process strings indicating options specified by the user. Use of
+        // any of these should prevent the "Add new torrent" dialog from
+        // opening:
+
+        if (param.startsWith("@savePath=")) {
+            torrentParams.savePath = param.mid(10);
+            skipTorrentDialog = true;
```

Comment:
Use TriStateBool for skipDialog is a really good idea. And, of course, we should implement overriding of AddNewTorrentDialog initial values (IMO, it should be done before these changes go to release).